eudemo/eudemo/equilibria/_equilibrium.py

```diff
@@ -94,3 +96,79 @@ def make_equilibrium(
     )
 
     return Equilibrium(coilset, grid, profiles)
+
+
+@dataclass
+class ReferenceEquilibriumParams(ParameterFrame):
+    """Parameters required to make a new reference equilibrium."""
+
+    A: Parameter[float]
+    B_0: Parameter[float]
+    I_p: Parameter[float]
+    kappa: Parameter[float]
+    R_0: Parameter[float]
+    r_cs_in: Parameter[float]
+    g_cs_mod: Parameter[float]
+    tk_cs_casing: Parameter[float]
+    tk_cs_insulation: Parameter[float]
+    tk_cs: Parameter[float]
+    beta_p: Parameter[float]
+    l_i: Parameter[float]
+    n_CS: Parameter[int]
+    n_PF: Parameter[int]
+
+
+def make_reference_equilibrium(
+    _params: Union[ReferenceEquilibriumParams, Dict],
+    tf_track: BluemiraWire,
+    lcfs_shape: BluemiraWire,
+    p_prime: np.ndarray,
+    ff_prime: np.ndarray,
+    nx: int = 65,
+    nz: int = 65,
+):
+    """
+    Make a crude reference equilibrium, scaling coils and grid for a first pass
+    solve.
+    """
+    if isinstance(_params, dict):
+        params = ReferenceEquilibriumParams.from_dict(_params)
+    else:
+        params = _params
+
+    coilset = make_reference_coilset(
+        tf_track,
+        lcfs_shape,
+        r_cs=params.r_cs_in.value + 0.5 * params.tk_cs.value,
+        tk_cs=0.5 * params.tk_cs.value,
+        g_cs_mod=params.g_cs_mod.value,
+        tk_cs_casing=params.tk_cs_casing.value,
+        tk_cs_insulation=params.tk_cs_insulation.value,
+        n_CS=params.n_CS.value,
+        n_PF=params.n_PF.value,
+    )
+
+    grid = make_grid(
+        params.R_0.value,
+        params.A.value,
+        params.kappa.value,
+        scale_x=1.5,
+        scale_z=1.6,
+        nx=nx,
+        nz=nz,
+    )
+
+    profiles = CustomProfile(
+        p_prime,
+        ff_prime,
+        R_0=params.R_0.value,
+        B_0=params.B_0.value,
+        I_p=params.I_p.value * 1e6,  # This is in fact required, James.
```

Comment:
Is it always the case that you want I_p in `μA`, or 1e6 times larger than the input?

If so this factor should probably go into CustomProfile as below if all user facing functions are in SI units. Otherwise can we use the below:
 ```suggestion
        I_p=raw_uc(params.I_p.value, "A", "μA")
```

Reply:
It should be in [A] (`params.I_p.unit' should only ever be amps because we autoconvert) , if thats the case the PLASMOD API needs to change, not adding a factor here.

Reply:
If everything is working correctly we already convert it: https://github.com/Fusion-Power-Plant-Framework/bluemira/blob/develop/bluemira/codes/plasmod/mapping.py#L579

Reply:
Aight sorry it was coming from I_p in mock_PROCESS being set in MA instead of A.